project/src/test/java/com/google/sps/SearchDataServletTest.java

```diff
@@ -58,45 +143,120 @@ public void setUp() throws Exception {
+    Mockito.when(getResponse.getWriter()).thenReturn(getPrintWriter);
+
+    postStringWriter = new StringWriter();
+    postPrintWriter = new PrintWriter(postStringWriter);
+    Mockito.when(postResponse.getWriter()).thenReturn(postPrintWriter);
+
+    secondPostStringWriter = new StringWriter();
+    secondPostPrintWriter = new PrintWriter(secondPostStringWriter);
+    Mockito.when(secondPostResponse.getWriter()).thenReturn(secondPostPrintWriter);
 
     searchDataServlet = new SearchDataServlet();
+    eventCreationServlet = new EventCreationServlet();
+    mockKeywordHelper = Mockito.mock(KeywordHelper.class);
+    eventCreationServlet.setSearchStore(new SearchStore(mockKeywordHelper));
+
+    // Mock a request to trigger the SpannerClient setup to run
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+    authenticationHelper.setUp();
+
+    // Mocking necessary for labels feature
+    NlpProcessing nlpProcessor = PowerMock.createMock(NlpProcessing.class);
+	PowerMock.expectNew(NlpProcessing.class).andStubReturn(nlpProcessor);
+	expect(nlpProcessor.getNlp(anyObject(String.class))).andStubReturn(new ArrayList());
+    PowerMock.replay(nlpProcessor, NlpProcessing.class);
+
+    SpannerTasks.insertOrUpdateUser(HOST);
+    loginHost();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    SpannerTestTasks.cleanup();
+    authenticationHelper.tearDown();
   }
 
+  /**
+   * Add event using eventCreationServlet and check that search for keyword using searchDataServlet
+   * not relevant in the event name or description does not appear in the results.
+   */
   @Test
-  public void addEventAndRetrieveResultsForKeywordNotInEventDescription_noResultsReturned()
+  public void oneEvent_KeywordNotRelevantInEventTitleOrDescription_noResultsReturned()
+      // ID         |   Title Has Games  |   Description Has Games
+      // 1                    No                  No
       throws IOException {
+    Mockito.when(postRequest.getParameter(PARAMETER_NAME))
+        .thenReturn(NAME_WITHOUT_GAMES);
     Mockito.when(postRequest.getParameter(PARAMETER_DESCRIPTION))
-        .thenReturn(DESCRIPTION_WITHOUT_CAKE_KEYWORD);
-    Mockito.when(postRequest.getParameter(PARAMETER_EVENT_ID)).thenReturn(EVENT_ID_1);
+        .thenReturn(DESCRIPTION_WITHOUT_GAMES);
+    setRequiredRequestParameters(postRequest);
     Mockito.when(getRequest.getParameter(PARAMETER_KEYWORD)).thenReturn(CAKE);
 
-    searchDataServlet.doPost(postRequest, postResponse);
+    // On the first call the getKeywords for title, return empty list
+    // On the second call the getKeyords for description, return empty list
+    Mockito.when(mockKeywordHelper.getKeywords())
+        .thenReturn(KEYWORDS_NAME_WITHOUT_GAMES, KEYWORDS_DESCRIPTION_WITHOUT_GAMES);
+
+    eventCreationServlet.doPost(postRequest, postResponse);
```

Comment:
Add comments for this part the the following part.

Reply:
Done.